ykrt/pass_finder/cargo_run.py

```diff
@@ -1,22 +1,22 @@
-import subprocess
-import time
-import sys, os
+import os, subprocess, time
 
 def run_test(yk_path,env=None, n=1) :
     prelink_passes = (env or os.environ).get('PRELINK_PASSES', 'Not Set')
     print(f"\033[93mPRELINK_PASSES: {prelink_passes}\033[0m")
     os.chdir("/home/shreei/research/yk_pv")
+    #subprocess.run(["cargo build"], shell=True, env=env or os.environ)
     times = []
     for i in range(n):
         before = time.time() 
-        try:
-            c = subprocess.run(["timeout 60 cargo test"], shell=True, env=env or os.environ) 
-        except:
-            print(f"\033[92m @@@@@@@@@@@@@@\033[0m")
+        c = subprocess.run(["timeout 60 cargo test"], shell=True, env=env or os.environ) 
```

Comment:
Ah, I think I see the challenge! `cargo test` is building Rust code with normal (non-ykllvm) LLVM.

What we want to benchmark is, for example, yklua running some benchmarks, because yklua is compiled with ykllvm.

Reply:
Can we rely on yklua's test for this? I remember some of the tests were running fine but always end with a SEG fault.

Reply:
In your case, you actually don't want the JIT to kick in: you want to just test the interpreter. So you'll want to set `yk_mt_hot_threshold_set` to a ludicrously high value (e.g. `1<<28` or something) in yklua. Then you should hopefully find that most benchmarks work fine. @Pavel-Durov can tell you at least a few which are known to work (even with the JIT on!).